src/vs/workbench/contrib/notebook/common/notebookEditorInput.ts

```diff
@@ -237,6 +245,39 @@ export class NotebookEditorInput extends AbstractResourceEditorInput {
 			}).join(', ');
 			throw new Error(`File name ${target} is not supported by ${provider.providerDisplayName}.\n\nPlease make sure the file name matches following patterns:\n${patterns}`);
 		}
+		// --- Start Positron ---
+		// When an untitled notebook is saved, we need to preserve the active runtime session
+		// By updating the session's URI reference, we maintain:
+		//  1. All defined variables and in-memory state
+		//  2. The active kernel connection
+		//  3. Execution history and context
+		// When saving an untitled notebook to disk, we must update the runtime session mappings
+		// to ensure the session (variables, execution history) remains connected to the new file
+		if (this.hasCapability(EditorInputCapabilities.Untitled) && target) {
+			try {
+				this._logService.debug(`Reassigning notebook session URI: ${this.resource.toString()} → ${target.toString()}`);
+
+				// Call updateNotebookSessionUri on the runtime service
+				// This updates internal mappings and emits events that other components listen for
+				const sessionId = this._runtimeSessionService.updateNotebookSessionUri(this.resource, target);
+
+				if (sessionId) {
+					// Log success to aid debugging session transfer issues
+					this._logService.debug(`Successfully reassigned session ${sessionId} to URI: ${target.toString()}`);
+				} else {
+					// This is an expected case for notebooks without executed cells (no session yet)
+					this._logService.debug(`No session found to reassign for URI: ${this.resource.toString()}`);
+				}
+			} catch (error) {
+				// Why we catch but continue:
+				// 1. Session transfer is important but secondary to saving the file content
+				// 2. Failed session transfer shouldn't prevent the user from saving their work
+				// 3. In the worst case, the notebook will save but users may need to re-run cells
+				console.error('Failed to update notebook session URI during save:', error);
+				this._logService.error('Failed to reassign notebook session URI', error);
```

Comment:
Is it intentional to have both `console` and `logService` logs?

Reply:
Nope, a leftover from the initial working before I hooked up the logging.